internal/serialization/default_portable_writer.go

```diff
@@ -189,7 +189,7 @@ func (pw *DefaultPortableWriter) setPosition(fieldName string, fieldType int32)
 	pos := pw.output.Position()
 	pw.output.PWriteInt32(pw.offset+field.Index*Int32SizeInBytes, pos)
 	pw.output.WriteInt16(int16(len(fieldName)))
-	pw.output.WriteBytes(fieldName)
+	pw.output.WriteStringBytes(fieldName)
```

Comment:
I've just changed the name of the function here, the string is iterated as if it is a byte array.

Reply:
I tried to run a code like this before writing this comment and before and after produced different byte arrays for out-of-extended-ASCII range strings

https://play.golang.org/p/VCSEn2Rq_VO

Reply:
I've added a test that checks for non-ASCII portable field names, It seems the prior code was wrong as well, since it wrote the length different than expected: https://github.com/hazelcast/hazelcast-go-client/pull/581/commits/fa87563ed814cbacaeec43b083ffc9427545e32b and https://github.com/hazelcast/hazelcast-go-client/pull/581/commits/41fae0df5b507d28b103ecd24bf2002ea023dde1